File: java/hardcoded_secrets/AuthConfig.java
import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import java.util.Date;
import java.util.Properties;
import jakarta.mail.*;
import jakarta.mail.internet.*;

public class AuthConfig {

    // VULN 1: Hardcoded JWT secret for token signing
    public String generateAuthToken(String userId, String role) {
        String jwtSecret = "shopist-super-secret-jwt-key-do-not-share-2024";
        return Jwts.builder()
            .setSubject(userId)
            .claim("role", role)
            .setIssuedAt(new Date())
            .setExpiration(new Date(System.currentTimeMillis() + 86400000L))
            .signWith(SignatureAlgorithm.HS256, jwtSecret.getBytes())
            .compact();
    }

    // VULN 2: Hardcoded SMTP credentials for order confirmation emails
    public void sendOrderConfirmationEmail(String toAddress, String orderSummary) throws MessagingException {
        String smtpUser = "shopist-noreply@shopist.com";
        String smtpPassword = "Sh0pist!Mail#2024";
        Properties props = new Properties();
        props.put("mail.smtp.host", "smtp.sendgrid.net");
        props.put("mail.smtp.port", "587");
        props.put("mail.smtp.auth", "true");
        props.put("mail.smtp.starttls.enable", "true");
        Session session = Session.getInstance(props, new Authenticator() {
            protected PasswordAuthentication getPasswordAuthentication() {
                return new PasswordAuthentication(smtpUser, smtpPassword);
            }
        });
        Message message = new MimeMessage(session);
        message.setFrom(new InternetAddress(smtpUser));
        message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(toAddress));
        message.setSubject("Your Shopist Order Confirmation");
        message.setText(orderSummary);
        Transport.send(message);
    }

    // VULN 3: Hardcoded admin credentials in login check
    public boolean isAdminUser(String username, String password) {
        String adminUsername = "shopist_admin";
        String adminPassword = "Adm1n@Shopist#Prod";
        return adminUsername.equals(username) && adminPassword.equals(password);
    }
}

File: kotlin/hardcoded_secrets/AuthConfig.kt
import com.auth0.jwt.JWT
import com.auth0.jwt.algorithms.Algorithm
import java.util.Properties
import jakarta.mail.Session
import jakarta.mail.internet.MimeMessage

class AuthConfig {

    // VULN 1: Hardcoded JWT secret in Algorithm.HMAC256 - token signing
    fun createJwtVerifier() = JWT.require(Algorithm.HMAC256("shopist_jwt_secret_do_not_share"))
        .withIssuer("shopist.com")
        .build()

    fun signToken(userId: String): String = JWT.create()
        .withIssuer("shopist.com")
        .withSubject(userId)
        .sign(Algorithm.HMAC256("shopist_jwt_secret_do_not_share"))

    // VULN 2: Hardcoded SMTP password in Session properties - order confirmation emails
    fun createMailSession(): Session {
        val props = Properties().apply {
            put("mail.smtp.host", "smtp.shopist.com")
            put("mail.smtp.port", "587")
            put("mail.smtp.auth", "true")
            put("mail.smtp.starttls.enable", "true")
            put("mail.smtp.user", "notifications@shopist.com")
            put("mail.smtp.password", "Sh0p1st_Sm7p_P@ss#2024!")
        }
        return Session.getInstance(props)
    }

    // VULN 3: Hardcoded admin credentials - fallback admin access
    fun isAdminCredentials(username: String, password: String): Boolean {
        val adminUsername = "shopist_admin"
        val adminPassword = "Adm1n$hop1st!BackD00r"
        return username == adminUsername && password == adminPassword
    }
}
